Doc: Arista AVD A-Care TAC Support Overview (#5136)
aristanetworks/ansible-avd: docs/support/support_overview.md

<!--
  ~ Copyright (c) 2025 Arista Networks, Inc.
  ~ Use of this source code is governed by the Apache License 2.0
  ~ that can be found in the LICENSE file.
  -->

# Arista AVD A-Care TAC Support Overview

> Arista AVD is a multi-domain network automation framework consumable as code, providing a continuous design framework. The AVD project is offered as an open-source project and backed by the world-class Arista Technical Support team through A-Care Service. TAC support for AVD must be purchased separately; see the ordering information below.

Reviewer: Arista AVD is a network automation framework covering multiple network design domains. AVD is an open-source project maintained by a dedicated Arista engineering team. Arista AVD customers have an option of ordering TAC support through A-Care Service. TAC support AVD must be purchased separately; see the ordering information below. Without support contract AVD code can be used on as-is basis, without any warranty and with best effort problem resolution via GitHub issues.

Not insisting on this description, but I'd certainly change the starting line.

Reviewer 2: I agree with Petr, plainer language is better. Just minor corrections:

Arista AVD is a network automation framework covering multiple network design domains. AVD is an open-source project maintained by a dedicated Arista engineering team. Customers have the option of purchasing TAC support for AVD through the A-Care Service. TAC support for AVD must be purchased separately; see the ordering information below. Without a support contract AVD code can be used as-is, without any warranty and with best effort problem resolution via GitHub issues.

Author: Thanks for the recommendation. I rephrased it!

## Support Offering

- AVD software is covered by [A-Care Priority Levels](https://www.arista.com/assets/data/tac/downloads/SRPriorityLevels.pdf).

> - TAC support covers software defects and troubleshooting Q&A.

Reviewer: defects -> bugs

Author: Let's stay with defects:

- In formal software engineering, quality assurance, and technical documentation, "defect" is generally considered the more precise and professional term.
- In everyday communication among developers or when speaking to a less technical audience, "bug" is perfectly acceptable and often more readily understood.

> - Provide comprehensive software lifecycle policy for AVD according to the AVD software life cycle policy detailed below.

Reviewer: detailed -> described

Reviewer 2: detailed is fine imo

Author: both are good, I'll stick with detailed!

I asked gemini for it's opinion:

- "Detailed" often suggests a higher level of specificity and granularity. If the section below truly goes into fine points and specifics, "detailed" might be slightly more precise.
- "Described" is slightly more general but perfectly suitable. It clearly indicates that an explanation follows.

- arista.avd
- arista.cvp

!!! note
    Red Hat supports “ansible-core” and Ansible Automation Platform. For non-AVD Ansible issues, please contact Red Hat Ansible TAC.

## Software Life Cycle Policy

Arista AVD Software Release Policy and Life Cycle defines the various phases of development and support to guide customers in transitioning to newer versions of the product based on the milestones in the life cycle. Arista Networks will support each major software release train (i.e., 4.x.x, 5.x.x) during the **Active Development** phase and up to 12 months after the release enters the **Maintenance** and **Support Only** phase. The following diagram depicts the release phases and the Arista TAC support mapping across this timeline.

### Software Life Cycle

![Figure: Arista AVD Software Life Cycle](../_media/software_lifecycle_light.svg#only-light)
![Figure: Arista AVD Software Life Cycle](../_media/software_lifecycle_dark.svg#only-dark)

**Active Development Phase:**

- TAC support available.
- Major release with new features, functionality, and bug fixes.

**Maintenance Phase:**

- TAC support available.
- Bug fixes on previous major release.

**Support Only Phase:**

> - Ongoing TAC support.

Reviewer 2: Why do we say "Ongoing TAC support" here and not simply "TAC support available" as above?

Author: resolved

- Software upgrade required for bug fixes.

### Example Release Timeline

The Arista AVD project follows [Semantic Versioning](../versioning/semantic-versioning.md): <font class="v-r">Major</font>.<font class="v-gr">Minor</font>.<font class="v-ygr">Maintenance</font> (ex. <font class="v-r">4</font>.<font class="v-gr">10</font>.<font class="v-ygr">2</font>):

> - <font class="v-r">Major: Contains breaking changes, follow the</font> [porting guide](../porting-guides/5.x.x.md).

Reviewer: may contain

Author: By definition, a Major release always contains breaking changes. The breaking changes may not apply to your deployment. I don't think we should be softer on language here.

<!--
  ~ Copyright (c) 2025 Arista Networks, Inc.
  ~ Use of this source code is governed by the Apache License 2.0
  ~ that can be found in the LICENSE file.
  -->

# Arista AVD A-Care TAC Support Overview

Arista AVD is a network automation framework covering multiple network design domains. AVD is an open-source project maintained by a dedicated Arista engineering team. Customers can purchase TAC support for AVD through the A-Care Service contract. TAC support for AVD must be purchased separately; see the ordering information below.

!!! note

>     Without a support contract AVD code can be used as-is, without any warranty and with best-effort problem resolution via [Github discussions board](https://github.com/aristanetworks/avd/discussions).

Reviewer 3: ```suggestion
    Without a support contract AVD code can be used as-is, without any warranty and with best-effort problem resolution via [GitHub discussions board](https://github.com/aristanetworks/avd/discussions).
```

## Support Offering

- AVD software is covered by [A-Care Priority Levels](https://www.arista.com/assets/data/tac/downloads/SRPriorityLevels.pdf).
- TAC support covers software defects and troubleshooting Q&A.

> - Provide comprehensive software lifecycle policy for AVD according to the AVD software life cycle policy detailed below.

Reviewer 4: ```suggestion
- Provide comprehensive software life cycle policy for AVD according to the AVD software life cycle policy detailed below.
```

- <font class="v-r">Major: Contains breaking changes, follow the</font> [porting guide](../porting-guides/5.x.x.md).
- <font class="v-gr">Minor: New features and fixes (non-breaking)</font>.
- <font class="v-ygr">Maintenance: Fixes only (non-breaking)</font>.

![Figure: Release timeline example](../_media/release_timeline_example_light.svg#only-light)
![Figure: Release timeline example](../_media/release_timeline_example_dark.svg#only-dark)

## Supported Upgrade Paths

AVD supports the following upgrade paths:

- Between any minor versions in the same release train, i.e., from 4.1.0 to 4.5.0.
  - Skipping minor versions is supported.
- Between major release trains, i.e., from 4.10.2 to 5.0.0.
  - Skipping a major version is not supported, i.e, from 3.8.0 to 5.0.0.
  - Major versions may contain changes requiring updates to the inventory variables. Required updates will be described in a Porting Guide for each major version.

Release notes and porting guide for version 5.x.x can be found here:

- [Release Notes](../release-notes/5.x.x.md)
- [Porting Guide](../porting-guides/5.x.x.md)

## Support Matrix

> The following table depicts the Arista Validated Designs release support matrix, including the timelines for each major software train (based on the lifecycle policy) and  the current state of support for each train.

Reviewer 4: ```suggestion
The following table depicts the AVD release support matrix, including the timelines for each major software train (based on the life cycle policy) and  the current state of support for each train.
```

| Release | Current Status | Initial Release Date | End of Support |
| ------- | -------------- | -------------------- | -------------- |
| 5.x.x | Active Development | Oct-2024 | TBD |

> | 4.x.x | Maintenance Phase | Jun-2023 | Oct-2025 |

Author: ```suggestion
| 4.x.x | Support Only Phase | Jun-2023 | Oct-2025 |
```

Author: @ClausHolbechArista @gmuloc - 4.x.x is entering support only phase correct?
